authz/providers/azure/rbac/rbac.go

```diff
@@ -443,33 +449,50 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 	}
 
 	// Fallback to managed namespace check
-	if !a.useManagedNamespaceResourceScopeFormat || a.clusterType != managedClusters {
-		klog.V(7).Infof("Checkaccess request is denied for user %s", checkAccessUsername)
-		return status, nil
-	}
+	managedNamespaceExists, managedNamespacePath := getManagedNameSpaceScope(request)
+	if a.useManagedNamespaceResourceScopeFormat &&
+		a.clusterType == managedClusters &&
+		managedNamespaceExists {
+		klog.V(7).Infof("Falling back to checking managed namespace scope for user %s", checkAccessUsername)
+		// Build managed namespace URL
+		managedNamespaceURL, err := buildCheckAccessURL(*a.apiURL, a.azureResourceId, true, managedNamespacePath)
+		if err != nil {
+			return nil, errors.Wrap(err, "error in building managed namespace check access URL")
+		}
 
-	klog.V(7).Infof("Falling back to checking managed namespace scope for user %s", checkAccessUsername)
-	exists, managedNamespacePath := getManagedNameSpaceScope(request)
-	if !exists {
-		klog.V(7).Infof(
-			"Skipping managed namespace check for user %s because subject access review is cluster scoped",
-			checkAccessUsername,
-		)
-		return status, nil
-	}
+		// Update resource IDs for managed namespace
+		for _, b := range checkAccessBodies {
+			b.Resource.Id = path.Join(a.azureResourceId, managedNamespacePath)
+		}
 
-	// Build managed namespace URL
-	managedNamespaceURL, err := buildCheckAccessURL(*a.apiURL, a.azureResourceId, true, managedNamespacePath)
-	if err != nil {
-		return nil, errors.Wrap(err, "error in building managed namespace check access URL")
+		status, err = a.performCheckAccess(managedNamespaceURL, checkAccessBodies, checkAccessUsername)
+		if err != nil {
+			return nil, err
+		}
+		if status != nil && status.Allowed {
+			klog.V(7).Infof("Managed namespace checkaccess request is allowed for user %s", checkAccessUsername)
+			return status, nil
+		}
 	}
 
-	// Update resource IDs for managed namespace
-	for _, b := range checkAccessBodies {
-		b.Resource.Id = path.Join(a.azureResourceId, managedNamespacePath)
+	// Fallback to fleet scope check when the managedCluster has joined a fleet
+	if found, fleetMgrID := a.getFleetManagerResourceID(); found {
+		fleetURL, err := buildCheckAccessURL(*a.apiURL, fleetMgrID, managedNamespaceExists, managedNamespacePath)
+		if err != nil {
+			return nil, errors.Wrap(err, "error in building fleet manager check access URL")
+		}
+		bodiesForFleetRBAC, err := prepareCheckAccessRequestBody(request, fleetmembers, fleetMgrID, false, a.allowCustomResourceTypeCheck)
```

Comment:
is fleet Managed Namespace tracked ARM resource? If it is tracked, could you make sure that the special logic in ([Support tracked managed namespace deletions by artunduman · Pull Request #414 · kubeguard/guard](https://github.com/kubeguard/guard/pull/414)) to cover 404 for managed namespaces of single cluster also covers fleet managed namespace.